src/collective/behavior/amp/browser/__init__.py

```diff
@@ -232,6 +234,12 @@ def sticky_ad(self):
         return getattr(self.settings, 'amp_sticky_ad', None)
 
     @property
+    def pixel(self):
+        adapter = queryAdapter(self.context, IAMPPixelProvider)
```

Comment:
If you follow my suggestion to change from simple adapter to view then this would look like:

`return api.content.get_view('amp-tracking-pixel', self.context, self.request())`

Implement a default `amp-tracking-pixel` view which renders an empty string in order to make the above code always work. I like this "sentinel" or "null value" pattern to avoid `if` statements.